lib/ae_mdw/contract.ex

```diff
@@ -310,6 +314,24 @@ defmodule AeMdw.Contract do
     Enum.group_by(get_events(micro_block), fn {_event_name, %{tx_hash: tx_hash}} -> tx_hash end)
   end
 
+  @spec get_txi(pubkey()) :: integer()
+  def get_txi(contract_pk) do
+    case :ets.lookup(:ct_create_sync_cache, contract_pk) do
```

Comment:
As this `get_txi` function is related to sync, it would be better to keep in previous module (`AeMdw.Db.Sync.Contract`)

Reply:
I think overall all of the resource modules that are under the `AeMdw.Db.Sync.*` should be removed, and instead moved to the regular modules `AeMdw.Db.*` (or maybe even `AeMdw.*`). Right now both `Db.Sync.*` and `Db.*` contain similar functions for all resources, and you never know where to place them. 

What do you think? Shouldn't they all belong to a single `Contract/Tx/Oracle/etc` module? If not, how do you know where to place them?

Reply:
I think the current division can be improved specially regarding the AEX9 but for scope of this PR we shouldn't change it. Ideally making it more clear what is specific for Sync and for Endpoints (besides separating what is specific for AEX9).